hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -181,6 +184,44 @@ def test_function(self, data):
         # Record the test result in the tree, to avoid unnecessary work in
         # the future.
 
+        # The tree has two main uses:
+
+        # 1. it is mildly useful in some cases during generation where there is
+        #    a high probability of duplication but it is possible to generate
+        #    many examples. e.g. if we had input of the form none() | text()
```

Comment:
This appears to be the start of a new sentence, so `e.g.` → `E.g.`.

Reply:
I believe that is a regional variation.